MPSGameLift/Scripts/export_gamelift_server_package.py

```diff
@@ -172,46 +179,54 @@
+def create_exe_package(new_package_folder_name, exe_name):
+    # Add all the required exe's, dll's, and asset pak files into a folder to upload to GameLift
+    package_cache_dir = os.path.join(new_package_folder_name, "Cache", "pc")
 
-# Delete the old server package
-if os.path.exists(gamelift_package_folder_name):
-    shutil.rmtree(gamelift_package_folder_name)
+    # Delete the old server package
+    if os.path.exists(new_package_folder_name):
+        shutil.rmtree(new_package_folder_name)
 
-# Create the folders
-os.makedirs(gamelift_package_cache_dir, exist_ok=True)
-os.makedirs(gamelift_package_gems_dir, exist_ok=True)
+    # Create the folders
+    os.makedirs(package_cache_dir, exist_ok=True)
+
+    # Copy .exe and .dll files to GameLiftWindowsServerPackage directory
+    build_dir = os.path.join(monolithic_build_folder, "bin", "release")
+    for file_name in os.listdir(build_dir):
+        file_path = os.path.join(build_dir, file_name)
+        if os.path.isfile(file_path) and file_name.lower().endswith((exe_name.lower(), '.dll')):
+            shutil.copy(file_path, new_package_folder_name)
 
-# Copy .exe and .dll files to GameLiftWindowsServerPackage directory
-build_dir = os.path.join(monolithic_build_folder, "bin", "release")
-for file_name in os.listdir(build_dir):
-    file_path = os.path.join(build_dir, file_name)
-    if os.path.isfile(file_path) and file_name.lower().endswith(('.exe', '.dll')):
-        shutil.copy(file_path, gamelift_package_folder_name)
+    # Copy .pak files to Cache\pc directory
+    for file_name in os.listdir(bundles_directory):
+        if file_name.endswith(".pak"):
+            file_path = os.path.join(bundles_directory, file_name)
+            shutil.copy(file_path, package_cache_dir)
 
-# Copy launch_server.cfg file to GameLiftWindowsServerPackage directory
+# Create the GameLift server package
+create_exe_package(gamelift_package_folder_name, 'ServerLauncher.exe')
+
+# GameLift server package needs launch_server.cfg file
 launch_server_cfg_filepath = os.path.join(o3de_context.project_path, "launch_server.cfg")
 if os.path.isfile(launch_server_cfg_filepath):
     shutil.copy(launch_server_cfg_filepath, gamelift_package_folder_name)
 else:
     o3de_logger.error(f"Could not find serverlauncher.cfg! Launch_server.cfg is required because there's a bug with multiplayer when calling --loadlevel in the command-line. See https://github.com/o3de/o3de/issues/15773.")
     quit()
 
-# Copy .pak files to Cache\pc directory
-for file_name in os.listdir(bundles_directory):
-    if file_name.endswith(".pak"):
-        file_path = os.path.join(bundles_directory, file_name)
-        shutil.copy(file_path, gamelift_package_cache_dir)
-
-# Copy files to Gems\AWSCore directory
+# GameLift server needs Gems\AWSCore
+gamelift_package_gems_dir = os.path.join(gamelift_package_folder_name, "Gems", "AWSCore")
+os.makedirs(gamelift_package_gems_dir, exist_ok=True)
```

Comment:
What specifically from the AWSCore is needed by Gamelift?

Because CMake post build command should be used to copy whatever that is to the build directory.

Reply:
Oh, I see, the Gems/AWSCore folder is the one that copied to the build directory that contains the `resource_mapping_schema.json` file.

Could you update the comment to mention that the AWS Core metadata files that have been output to the build directory is what is being copied to the layout.

Reply:
Fixed!